modules/webgl/src/adapter/resources/webgl-buffer.ts

```diff
@@ -60,7 +60,11 @@ export class WEBGLBuffer extends Buffer {
   // PRIVATE METHODS
 
   /** Allocate a new buffer and initialize to contents of typed array */
-  _initWithData(data, byteOffset: number = 0, byteLength: number = data.byteLength + byteOffset): this {
+  _initWithData(
+    data,
+    byteOffset: number = 0,
+    byteLength: number = data.byteLength + byteOffset
+  ): this {
     assert(ArrayBuffer.isView(data));
```

Comment:
I'm pretty sure this should be `byteLength = data.byteLength` and not `byteLength = data.byteLength + byteOffset`, anything I'm missing?

Reply:
Oops, I misunderstood the Buffer API here. The `byteOffset` defines _padding_ inserted at the start of the buffer, after which comes the full contents of the data array. So the full byte length of the buffer is really `data.byteLength + byteOffset`.

I'll leave this alone in that case.